Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -984,7 +989,7 @@ struct PluginScriptRunnerInputSerializer {
                 major: package.manifest.toolsVersion.major,
                 minor: package.manifest.toolsVersion.minor,
                 patch: package.manifest.toolsVersion.patch),
-            dependencies: try package.dependencies.map{ .init(packageId: try serialize(package: $0)) },
```

Comment:
I fail to see the difference between this and the fix above, can you help me see it?

Reply:
This moves the serialization of the other packages up before the next wire ID is assigned, so that it refers to the correct package.  There are more comments about how this works in this source file but I'll add some commentary here to make this clear.

Reply:
Done